deepchecks/recommender/ranking.py

```diff
@@ -7,705 +7,423 @@
+        items_popularity (Union[dict, None]): Item popularity scores.
+        num_users (int): Total number of users.
+        k (int): The number of top items to consider for novelty. Default is 20.
 
     Returns:
-        (float): novelty
-
-    Metric Definition:
-    Zhou, T., Kuscsik, Z., Liu, J. G., Medo, M., Wakeling, J. R., & Zhang, Y. C. (2010).
-    Solving the apparent diversity-accuracy dilemma of recommender systems.
-    Proceedings of the National Academy of Sciences, 107(10), 4511-4515.
-
-    `Original <https://github.com/statisticianinstilettos/recmetrics/blob/master/recmetrics/metrics.py#L14>`_
-    """
-    if item_popularity is None:
-        raise ValueError("item_popularity must be provided.")
-
-    epsilon = 1e-10
+        Tuple[float, float]: The popularity-based novelty score and the normalized score.
+    """
+    #epsilon = 1e-10
+    # Calculate the mean surprisal/self-information of the top K popular head items
+    top_information_mean = 0
+    most_popular_item_catalog = round(0.05 * len(items_popularity))
+    popularity_topk_items = sorted(items_popularity)[-most_popular_item_catalog:]
+    for item_ in popularity_topk_items:
+        top_information_mean += -np.log2(( item_ + 1e-10) / num_users)
+    top_information_mean = top_information_mean / most_popular_item_catalog
     all_self_information = []
-    self_information_sum = 0.0
-    for i in range(len(recommendation)):
-        item = recommendation[i]
-        item_pop = item_popularity[item]
-        self_information_sum += -log2((item_pop + epsilon) / num_users)
+    for i,rec in recommendations:
+        self_information_sum = 0.0
+        for i in range(k):
+            item_pop = items_popularity[rec[i]]
+            self_information_sum += -np.log2((item_pop + 1e-10) / num_users)
 
-    avg_self_information = self_information_sum / k
+        avg_self_information = self_information_sum / k
+        all_self_information.append(avg_self_information)
 
-    return avg_self_information
+    # Calculate novelty and normalized novelty
+    novelty = np.mean(all_self_information)
+    normalized_novelty = novelty / top_information_mean
+    return novelty, normalized_novelty
 
-
-def normalized_distance_based_performance_measure(relevant_items: List, recommendation: List):
+def diversity_score(predictions: List, target_vectors: Union[List, np.array],
+                    top_n: int = 10) -> float:
     """
-    Calculates the Normalized Distance-based Performance Measure (NPDM) between two
-    ordered lists. Two matching orderings return 0.0 while two unmatched orderings returns 1.0.
-
-    Args:
-        relevant_items (List): List of items
-        recommendation (List): The predicted list of items
+    Compute the diversity score of a set of recommendations.
+    Parameters:
+    predictions (List): List of predicted item indices for each user.
+    target_vectors (Union[List, np.array]): List or array of item feature vectors.
```

Comment:
maybe call this item_features in that case?

Reply:
fixed